src/Parser/ShadowPriest/Modules/Spells/Dispersion.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Module from 'Parser/Core/Module';
+
+import SpellLink from 'common/SpellLink';
+
+import { formatPercentage } from 'common/format';
+import { calculateMaxCasts } from 'Parser/Core/getCastEfficiency';
+
+class Disperion extends Module {
+  _dispersions = {};
+  _previousDispersionCast = null;
+
+  on_initialized() {
+    this.active = true;
+  }
+
+  get uptime(){
+    return this.dispersionsAsArray.reduce((p, c) => p += c.end - c.start, 0);
+  }
+
+  get maxUptime(){
+    const fightDuration = this.owner.fight.end_time - this.owner.fight.start_time;
+    return Math.round(calculateMaxCasts(90 - (10 * this.owner.selectedCombatant.traitsBySpellId[SPELLS.FROM_THE_SHADOWS_TRAIT.id]), fightDuration)) * 6000;
```

Comment:
Please do not use any *magic numbers* in the code and define them as a constant at the top of your file if they're only going to be used in one place and you don't need any additional info (e.g. `const DISPERSION_UPTIME_MS = 6000;`).